packages/zoe/test/unitTests/contracts/test-autoswap.js

```diff
@@ -220,113 +562,305 @@ test('autoSwap - test fee', async t => {
       zoe,
     } = setup();
-    const invitationIssuer = zoe.getInvitationIssuer();
     const installation = await installationPFromSource(zoe, autoswap);
 
-    // Setup Alice
-    const aliceMoolaPayment = moolaMint.mintPayment(moola(10000));
-    const aliceSimoleanPayment = simoleanMint.mintPayment(simoleans(10000));
-
-    // Setup Bob
-    const bobMoolaPayment = moolaMint.mintPayment(moola(1000));
-
-    // Alice creates an autoswap instance
+    // create an autoswap instance
     const issuerKeywordRecord = harden({
-      TokenA: moolaIssuer,
-      TokenB: simoleanIssuer,
+      Central: moolaIssuer,
+      Secondary: simoleanIssuer,
     });
     const { publicFacet } = await zoe.startInstance(
       installation,
       issuerKeywordRecord,
     );
     const liquidityIssuer = await E(publicFacet).getLiquidityIssuer();
     const liquidity = (await makeLocalAmountMath(liquidityIssuer)).make;
+    const kits = {
+      moolaIssuer,
+      simoleanIssuer,
+      liquidityIssuer,
+      moola,
+      simoleans,
+      liquidity,
+      zoe,
+    };
+    let poolState = {
+      m: 0,
+      s: 0,
+      l: 0,
+      k: 0,
+    };
 
-    // Alice adds liquidity
-    const aliceProposal = harden({
-      give: {
-        TokenA: moola(10000),
-        TokenB: simoleans(10000),
-      },
-      want: { Liquidity: liquidity(0) },
-    });
-    const alicePayments = harden({
-      TokenA: aliceMoolaPayment,
-      TokenB: aliceSimoleanPayment,
-    });
-
-    const aliceAddLiquidityInvitation = await publicFacet.makeAddLiquidityInvitation();
-    const aliceSeat = await zoe.offer(
-      aliceAddLiquidityInvitation,
-      aliceProposal,
-      alicePayments,
+    // Setup Alice
+    const moolaPurse = moolaIssuer.makeEmptyPurse();
+    moolaPurse.deposit(moolaMint.mintPayment(moola(50000)));
+    const simoleanPurse = simoleanIssuer.makeEmptyPurse();
+    simoleanPurse.deposit(simoleanMint.mintPayment(simoleans(50000)));
+    const alice = makeTrader(
+      moolaPurse,
+      simoleanPurse,
+      liquidityIssuer.makeEmptyPurse(),
+      kits,
+      publicFacet,
     );
 
-    assertOfferResult(t, aliceSeat, 'Added liquidity.');
+    const initLiquidityDetails = {
+      mAmount: moola(10000),
+      sAmount: simoleans(10000),
+    };
+    const initLiquidityExpected = {
+      m: 10000,
+      s: 10000,
+      l: 10000,
+      k: 100000000,
+      payoutL: 10000,
+      payoutM: 0,
+      payoutS: 0,
+    };
+    await alice.initLiquidityAndCheck(
+      t,
+      poolState,
+      initLiquidityDetails,
+      initLiquidityExpected,
+    );
+    t.assert(t, '..Alice added liquidity');
+    poolState = updatePoolState(poolState, initLiquidityExpected);
 
-    const liquidityPayout = await aliceSeat.getPayout('Liquidity');
+    const tradeDetails = {
+      inAmount: moola(1000),
+      outAmount: simoleans(0),
+    };
+    const tradeExpected = {
+      m: 11000,
+      s: 9094,
+      l: 10000,
+      k: 11000 * 9094,
```

Comment:
I think we shouldn't be calculating postK, but rather using >= preK as the invariant

Reply:
Seems like we are not testing as much if we hard-code postK based on the swap output that we get from our own code.